python/ray/_private/worker.py

```diff
@@ -2630,50 +2638,62 @@ def put(
 @PublicAPI
 @client_mode_hook
 def wait(
-    object_refs: List["ray.ObjectRef"],
+    ray_waitables: List[RayWaitable[Any]],
     *,
     num_returns: int = 1,
     timeout: Optional[float] = None,
     fetch_local: bool = True,
-) -> Tuple[List["ray.ObjectRef"], List["ray.ObjectRef"]]:
+) -> Tuple[List[RayWaitable[Any]], List[RayWaitable[Any]]]:
     """Return a list of IDs that are ready and a list of IDs that are not.
 
     If timeout is set, the function returns either when the requested number of
     IDs are ready or when the timeout is reached, whichever occurs first. If it
     is not set, the function simply waits until that number of objects is ready
     and returns that exact number of object refs.
 
-    This method returns two lists. The first list consists of object refs that
-    correspond to objects that are available in the object store. The second
-    list corresponds to the rest of the object refs (which may or may not be
-    ready).
+    `ray_waitables` is a list of :class:`~ObjectRefs` and
+    :class:`~StreamingObjectRefGenerators`.
+
+    ObjectRef:
+        The first list consists of object refs that
+        correspond to objects that are available in the object store.
+        The second list corresponds to the rest of the object refs
+        (which may or may not be ready).
 
-    Ordering of the input list of object refs is preserved. That is, if A
+    StreamingObjectRefGenerators:
+            The first list consists of generators in which the next ref
+            in the genreator correspond to objects that are
+            available in the object store.
+            The second list corresponds to the rest of the generator
+            in which the next item is not ready in the object store.
```

Comment:
Nit, this comment gives me a feeling that ObjectRef and StreamingObjectRefGenerators are inclusive. Maybe we can phrase the comment in this way: first say the first list contains ready waitables, and the second is unready; then explain what ready/unready means for different types.

Reply:
This is a nice suggestion. Can. you take a look at the revision?